vyper/semantics/types/utils.py

```diff
@@ -17,6 +17,13 @@
 from vyper.semantics.validation.utils import get_exact_type_from_node, get_index_value
 
 
+class KwargSettings(object):
```

Comment:
please add some comments explaining what this class is for. also, no need to specify that it inherits from `object` since that is the default for python classes
```suggestion
# convenience class which holds metadata about how to process kwargs.
# contains the `default` value for the kwarg as a python value, and a
# flag `require_literal`, which, when True, indicates whether or not,
# indicates that the kwarg must be set to a compile-time constant at any    
# call site. (note that the kwarg processing machinery will return a
# Python value instead of an AST or IRnode in this case).
class KwargSettings:
```